# CP-5 fix the eslint bug

samyshyaka opened the pull request:
#### What the PR does
Fix the Eslint bug

#### Description of Tasks to be completed

- Update the Eslint rules
- Restore the affected code in the src folder

#### How this should be manually tested
Clone the project repository linked [here](https://github.com/atlp-rwanda/phantom-fe-codebandits)

```
cd phantom-fe-codebandits/
git checkout ch-eslint-bug-fix-cp-5
git fetch origin
npm install
npm run start
run http://localhost:3000 in your browser
```

#### Relevant Trello Card
N/A

samyshyaka committed:
fixed the eslint bug

updated the husky git ingore files

updated the husky git ingore files

updated the husky git ingore files

updated the husky git ingore files

samyshyaka committed:
fixed the eslint bug

updated the husky git ingore files

updated the husky git ingore files

updated the husky git ingore files

updated the husky git ingore files

added the allow arrow function rule to eslint rules

samyshyaka committed:
Merge branch 'ch-eslint-bug-fix-cp-5' of https://github.com/atlp-rwanda/phantom-fe-codebandits into ch-eslint-bug-fix-cp-5

Yvad60 commented on .eslintrc.json:
```diff
@@ -17,6 +17,26 @@
+    "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx"] }],
+    "func-style": ["error", "declaration", { "allowArrowFunctions": true }]
+  },
+  "prettier/prettier": [
+    "error", 
+    {
+      "bracketSpacing": true,
+      "semi": true,
+      "singleQuote": true,
+      "trailingComma": "none",
+      "printWidth": 80,
+      "tabWidth": 2
+    }
+  ]
```
can you double-check these changes because end I'm getting  `Error: ESLint configuration in .eslintrc.json is invalid:
        - Unexpected top-level property "prettier/prettier".`  while running `npm run lint`  I don't know if it would be necessary to specify prettier rules again in `eslintrc.json` since we already have the same rules in  `.prettierrc` file

samyshyaka replied:
Forgot to remove those from the .prettierrc file but it seemed that the prettier extension of the vscode was not reading them at all. I found it better to put them in .eslintrc.json file. Let me revise this and find the perfect solution

Yvad60 replied:
about the extensions, I have seen that it is recommended to temporarily disable Eslint or prettier extensions to test that formatting fixes are actually happening because of the rules in `eslintrc.json` or `.prettierrc` not being done by extensions themselves. it is to make sure the rules will work in case someone doesn't have those extensions

Yvad60 reviewed (commented):
nice work, the error we were getting due to imports is fixed, and the project is running in the browser again but
confirm that the husky pre-commit hook is working for you. I had tested on my end and there was no pre-commit file in the husky folder. For testing, I intentionally added a file with formating and linting mistakes and I was able to commit without those mistakes fixed which indicate that linting pre-commit is not working

Yvad60 reviewed (commented):
in the description for this pull request change the checkout command to checkout to this branch not the redux one

samyshyaka commented:
> in the description for this pull request change the checkout command to checkout to this branch not the redux one

Sure

samyshyaka committed:
reconfigured eslint, prettier and husky

updated index.js

updated files after adding more eslint rules

Yvad60 commented on .eslintrc.json:
```diff
@@ -1,22 +1,34 @@
-  "rules": {
-    "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx"] }]
-  }
+    "extends": [
+        "plugin:react/recommended",
+        "airbnb"
+    ],
```
we are no longer extending any prettier dependencies in `eslintrc.json` therefore Eslint default formatting rules are the ones being used instead of using prettier for formatting 
I tested and prettier's rules only run only when you manually run `npm run format` but so far `npm run lint` doesn't consider prettier

Yvad60 replied:
> we are no longer extending any prettier dependencies in `eslintrc.json` therefore Eslint default formatting rules are the ones being used instead of using prettier for formatting I tested and prettier's rules only run only when you manually run `npm run format` but so far `npm run lint` doesn't consider prettier

the pre-commit script is now set up but the problem stated above remains and prettier rules are not being used for formatting before committing

feyton replied:
Can you elaborate on that and explain the issue

Yvad60 commented on package.json:
```diff
@@ -38,20 +35,24 @@
     "webpack-cli": "^4.9.2",
     "webpack-dev-server": "^4.7.4"
+  },
+  "lint-staged": {
+    "*.js": [
+      "eslint --cache --fix",
+      "prettier --write"
+    ]
```
confirm that this section is working as expected.
I think it was supposed to fix eslint and prettier formatting mistakes on .js staged files before committing but on my end, it's not running at all

Yvad60 reviewed (commented):
Good work man! the prettier error we were getting is fixed and `npm run lint` is now working and the server is running as expected in the browser. so far commits with files including mistakes are still able to pass I think the remaining part is to run eslint and prettier on each commit to prevent that

feyton reviewed (changes requested):
## Add a pre-commit script

The PR is missing the script to check for errors. and the prevents the commit if the style are not correct. After adding that script.

- [x] Add pre-commit script

feyton committed:
ch-(Eslint bug)- Add pre-commit script

This commit add the precommit script to check for linting issues before
committing. The edit to gitignore allow the pre-commit to be pushed to
github

samyshyaka committed:
updated pre-commit file permissions

Yvad60 committed:
extends prettier rules in eslintrc

Yvad60 committed:
restaging files after linting

this commit add the format script which will run prettier on every
staged file just in cases the commit includes other files like .css
files that may need formating
it also fix and restages files after linting

Yvad60 reviewed (approved):
Tested and eslint, prettier and  pre-commit scripts are running as expected with all required rules

feyton commented on .husky/pre-commit:
```diff
@@ -1,8 +1,9 @@
-)
\ No newline at end of file
+)
+git add .
```
This really good implementation

feyton reviewed (approved):
The PR work as expected and the issue was resolved
